thriftbp/client_middlewares.go

```diff
@@ -108,6 +110,24 @@ func SetDeadlineBudget(next thrift.TClient) thrift.TClient {
 	}
 }
 
+// Retry returns a thrift.ClientMiddleware that can be used to automatically
+// retry thrift requests.
```

Comment:
nit: Probably worth mentioning in the doc comment that this is not enabled by default in baseplate and service code need to add it explicitly.

Reply:
I was actually considering making it enabled by default, just with only 1 attempt >.>

Then you can still set per-call retry behavior if you want with the context function.

Reply:
Does "1 attempt" mean no retry or one retry?

Reply:
It means no retries: https://github.com/avast/retry-go/blob/a8f6dc7e8f46a5d11c01d9da1e887bb5a51fd107/retry.go#L113

Reply:
and just to double check, if the one and only attempt failed, it won't do any delay and will return the error directly, right? in that case make retry middleware default with one attempt is fine.

Reply:
Hmm, it looks like it doesn't do that, it will at least go _through_ the logic to see if it should "retry" and might do the delay if it does.

Reply:
https://github.com/avast/retry-go/blob/a8f6dc7e8f46a5d11c01d9da1e887bb5a51fd107/retry.go#L125-L128

Looks like we should be good?

Reply:
This should also be verifiable via a unit test. You just create a retry policy, then pointing the client to a non-existing host:port, make sure that error is retryable, set a large delay, and check whether the time of the call included the delay or not.

Reply:
Oh great! My eyes totally glazed over that part, I'll update this and add that.